tracking/include/pcl/tracking/pyramidal_klt.h

```diff
@@ -37,343 +37,408 @@
-               const PointCloudInConstPtr& current_input,
-               const std::vector<FloatImageConstPtr>& previous_pyramid,
-               const std::vector<FloatImageConstPtr>& current_pyramid,
-               const pcl::PointCloud<pcl::PointUV>::ConstPtr& previous_keypoints,
-               pcl::PointCloud<pcl::PointUV>::Ptr& current_keypoints,
-               std::vector<int>& status,
-               Eigen::Affine3f& motion) const;
-
-        void
-        computeTracking () override;
-
-        /// \brief input pyranid at t-1
-        std::vector<FloatImageConstPtr> ref_pyramid_;
-        /// \brief point cloud at t-1
-        PointCloudInConstPtr ref_;
-        /// \brief number of pyramid levels
-        int nb_levels_;
-        /// \brief detected keypoints 2D coordinates
-        pcl::PointCloud<pcl::PointUV>::ConstPtr keypoints_;
-        /// \brief status of keypoints of t-1 at t
-        pcl::PointIndicesPtr keypoints_status_;
-        /// \brief number of points to detect
-        std::size_t keypoints_nbr_;
-        /// \brief tracking width
-        int track_width_;
-        /// \brief half of tracking window width
-        int track_width_2_;
-        /// \brief tracking height
-        int track_height_;
-        /// \brief half of tracking window height
-        int track_height_2_;
-        /// \brief maximum number of iterations
-        unsigned int max_iterations_;
-        /// \brief accuracy criterion to stop iterating
-        float accuracy_;
-        float min_eigenvalue_threshold_;
-        /// \brief epsilon for subpixel computation
-        float epsilon_;
-        float max_residue_;
-        /// \brief number of hardware threads
-        unsigned int threads_;
-        /// \brief intensity accessor
-        IntensityT intensity_;
-        /// \brief is the tracker initialized ?
-        bool initialized_;
-        /// \brief compute transformation from successfully tracked points
-        pcl::TransformationFromCorrespondences transformation_computer_;
-        /// \brief computed transformation between tracked points
-        Eigen::Affine3f motion_;
-        /// \brief smoothing kernel
-        Eigen::Array<float, 5, 1> kernel_;
-        /// \brief smoothing kernel half size
-        int kernel_size_2_;
-        /// \brief index of last element in kernel
-        int kernel_last_;
-      public:
-        PCL_MAKE_ALIGNED_OPERATOR_NEW
-    };
+    accuracy_ = accuracy;
   }
-}
+
+  /** \return the accuracy */
+  inline float
+  getAccuracy() const
+  {
+    return (accuracy_);
+  }
+
+  /** Set epsilon
+   * \param[in] epsilon desired epsilon.
+   */
+  inline void
+  setEpsilon(float epsilon)
+  {
+    epsilon_ = epsilon;
+  }
+
+  /** \return the epsilon */
+  inline float
+  getEpsilon() const
+  {
+    return (epsilon_);
+  }
+
+  /** \brief Set the maximum number of points to track. Only the first
+   * keypoints_nbr_ are used as points to track after sorting detected keypoints
```

Comment:
Do we need special syntax to link this variable or is it picked up automatically by doxygen?

PS: It should be `number` since `keypoints_nbr_` is implementation detail

Reply:
This tiny change will trigger the whole pipeline :rofl:

Reply:
Your diff doesn't highlight `number` as a variable

Reply:
We don't use `\var`.  Isn't the `\param` sufficient?

Also, I mistyped number as numbers. I'll change that.

Reply:
IIRC, that doesn't work in the brief (where you mention the variable)

Reply:
Does it have to be picked up as variable? 
There is `\param[in] number the desired number of points to detect.` in the doxygen comment.

Idk, please do suggest what changes to be made.

Reply:
> Does it have to be picked up as variable?

Not really.  But the language becomes confusing unless I have read the param line which comes after it. I'd suggest rewording the brief and param to work well together

Reply:
This okay?
```
  /** \brief Set the maximum number of points to track.
   * \param[in] number the desired number of points to detect after sorting detected keypoints
   * according to their response measure.
   */
```

Reply:
Second :)

Reply:
for this PR, good enough :)